session.go

```diff
@@ -345,9 +336,6 @@ func (s *session) Retry() error {
 // Unlike normal Exec, it doesn't reset statement status, doesn't commit or rollback the current transaction
 // and doesn't write binlog.
 func (s *session) ExecRestrictedSQL(ctx context.Context, sql string) (ast.RecordSet, error) {
-	if err := s.checkSchemaValidOrRollback(); err != nil {
```

Comment:
Removing this code may be problematic. Some actions are inconsistent with MySQL.
```
begin;
sql1(it's fail to execute);
sql2;
commit;
```